src/game.cpp

```diff
@@ -3552,6 +3552,16 @@ void Game::processClientEvents(CameraOrientation *cam, float *damage_flash)
 			bool enable = event.override_day_night_ratio.do_override;
 			u32 value = event.override_day_night_ratio.ratio_f * 1000;
 			client->getEnv().setDayNightRatioOverride(enable, value);
+		} else if (event.type == CE_SET_CLOUDS) {
```

Comment:
This isn't really working, sorry. Less than 12 hours after I pushed, this pull request acquired a conflict with the master branch due to unrelated changes in game.cpp. I'm splitting the refactoring out to a separate pull request, where it's better suited than such a long-lived pull request as this one.